### ISSUE TITLE
[device-port] [s3ve3g] port to Samsung S3 Neo

### ISSUE THREAD

**Issue by Flohack74** [Halium/projectmanagement#41]
- [x] Manifest with kernel, vendor, device tree
- [x] Setup kernel defconfig for hybris-boot
- [x] Build `hybris-boot` and `system.img`
- [x] Flash and test with reference rootfs
    - [x] Have a working SSH session
    - [x] Get LXC working
    - [x] Get Wifi working
    - [x] Halium tests (All working tests)
        - [ ] test_audio           (Broken? Halium-5.1)
        - [ ] test_camera        (Broken? Halium-5.1)
        - [ ] test_egl               (Broken? Halium-5.1)
        - [x] test_egl_configs 
        - [ ] test_glesv2          (Broken? Halium-5.1)
        - [x] test_gps              
        - [x] EGL_PLATFORM=hwcomposer test_hwcomposer
        - [ ] test_input            (Broken? Halium-5.1)
        - [x] test_lights           
        - [ ] test_media          (Not tested)
        - [ ] test_nfc               (Broken? Halium-5.1)
        - [ ] test_recorder      (Broken? Halium-5.1)
        - [x] test_sensors      
        - [x] test_vibrator      
        - [ ] test_wifi              (Broken? Halium-5.1)
---
- **Manifest**
    - [ ] Local manifest
- **Sources**
    - [x] Kernel             <https://github.com/Flohack74/android_kernel_samsung_s3ve3g/tree/halium-5.1> 
    - [x] Device             <https://github.com/Flohack74/android_device_samsung_s3ve3g/tree/halium-5.1>
    - [x] Vendor             <https://github.com/TheMuppets/>
- **Prebuilts**
    - [ ] To be released

---

Testing rootfs:

- [x] Halium Reference rootfs
    - [x] Works (See Halium tests)
- [ ] Arch Linux rootfs
    - [ ] ssh works
- [ ] UBPorts rootfs
    - [ ]  Stays on Google image
    - [ ] SSH Server starts up
- [ ] Plasma Mobil rootfs
    - [ ] Wifi
    - [ ] Sound (Volume buttons)
    - [ ] Touchscreen
    - [ ] Powerbutton

**Comment by Flohack74** [comment]
Current manifest is still heavily WiP, so its just attached here:
[local_manifests.zip](https://github.com/Halium/projectmanagement/files/1624053/local_manifests.zip)
Also, this is a download mode device, expect issues with the usual procedures. You will need heimdall flash tool instead of fastboot. I will be happy to help :)

**Comment by vanyasem** [comment]
@Flohack74 use github gist for the manifest plz

**Comment by LNJ2** [comment]
Please add your manifest to https://github.com/Halium/halium-devices. Thank you! 😃

**Comment by Flohack74** [comment]
Finally got the device manifest uploaded hehe

**Comment by Flohack74** [comment]
I am stepping back all tests that fail this year. Seems a few things got messed up by itself. But I am back to booting and ssh. Thats already a bit.

**Comment by Flohack74** [comment]
![image20190206_232639721](https://user-images.githubusercontent.com/16186470/52378118-2149e600-2a67-11e9-9482-4894917e3665.jpg)

**Comment by hpagseddy** [comment]
@Flohack74 do RIL works on Ubports?

**Comment by Flohack74** [comment]
Not on S3 Neo but I did not spend much time looking for a solution.

**Comment by Flohack74** [comment]
But network works, just no calls...

**Comment by hpagseddy** [comment]
@Flohack74 what about SMS? I am planning to use it on my S3

**Comment by Flohack74** [comment]
I did not try this out, sorry.

**Comment by nobodywasishere** [comment]
I want to help get this running if I can, however I'm completely new to UB Touch. I already have a device setup with TWRP that I don't care about wiping either. Could you point me in the right direction for building for this device starting from the work you've already completed?

**Comment by Flohack74** [comment]
Well I recently started a new tryout with using Halium 7.1 - Problem here is, S3 Neo never got an official CM 14.1 port, so it might be the wrong way. So far its booting but the Android container crashes due to wrong vendor blobs or idk. I will update here.

On the other hand, Halium 5.1 - which I used for the previous attempt, is not updated with important fixes right now. So you could get to the point I documented above, but then you will be probably stuck. I need to find out from the Halium group, if it is ever intended to update 5.1 branches with stuff from 7.1 - however, I am afraid this will not happen for a long time, since 5.1 devices are very old, and might not get any interest by the community.

**Comment by nobodywasishere** [comment]
Does it have to be CM specifically? I know there's an unofficial LOS 14.1 on XDA and it just got official LOS 16.0 support as well.

**Comment by Flohack74** [comment]
Unofficial ports do carry the possibility of being hackish, and that might make porting even harder. Currently you can only port for an Android 5.1 or 7.1 based device, and that means CM 12.1 or 14.1 back in those days. LOS 16 does not help at all, sorry.

**Comment by ghost** [comment]
can I use this on a s3 international lte

**Comment by Flohack74** [comment]
@RadGade sry no, thats a totally different device.

**Comment by ghost** [comment]
@Flohack74 all good, looks like I gotta make a port also where is the best place to ask questions if run into problems during the port

**Comment by Flohack74** [comment]
Better not port it: It has Kernel 3.0 and thats wayyyy too old...

**Comment by luka177** [comment]
> Better not port it: It has Kernel 3.0 and thats wayyyy too old...

Yes for s3 mainline is better.
Butw dose plasma mobile work on s3 neo?
or kernel 3.4 is problem

**Comment by Flohack74** [comment]
I dont know of any successful port for the S3, sorry.

**Comment by luka177** [comment]
I mean s3 (not neo) hav good mainline support.
And i ask about plamo on ss NEO (s3ve3g), as you ported Halium. Mb you tryed that.

**Comment by Flohack74** [comment]
No, sorry, I focused only on Ubuntu Touch. Since Halium is not unified for PMOS and UT you basically need to port the device 2 times. They do not share the same kernel configuration, for example. Mainline support is well for S3, yes, but does still all Android hardware work with it? I have my doubts about using a mainline kernel. You will basically get gUI + WiFi, but network data + calls and anything else is very questionable.

**Comment by luka177** [comment]
> No, sorry, I focused only on Ubuntu Touch. Since Halium is not unified for PMOS and UT you basically need to port the device 2 times. They do not share the same kernel configuration, for example. Mainline support is well for S3, yes, but does still all Android hardware work with it? I have my doubts about using a mainline kernel. You will basically get gUI + WiFi, but network data + calls and anything else is very questionable.

Here you are wronk. Most hw work on mainline ok. Except barometr an bt

**Comment by Flohack74** [comment]
Well you are invited to do a port then ;)

**Comment by luka177** [comment]
Lol i dont have galaxy s3 (exynos, good mailine support), i have s3 neo (snapdragon, no mainline). So ...

**Comment by Flohack74** [comment]
Then you are invited to install my port and play a bit with it... ^^
